librz/util/bitvector.c

```diff
@@ -200,6 +200,137 @@ RZ_API ut32 rz_bv_copy(RZ_NONNULL const RzBitVector *src, RZ_NONNULL RzBitVector
 	return dst->_elem_len;
 }
 
+/**
+ * \brief Optimized version of rz_bv_copy_nbits() for small bitvectors (64 or less bits)
+ *
+ * Copies `nbit` bits from `src_start_pos` to `dst_start_pos` using bitwise operations.
+ * This function is intended for bitvectors that fit within 64 bits and performs the copy
+ * in a single operation rather than bit-by-bit.
+ *
+ * \param src RzBitVector, data source (must be 64 or less bits)
+ * \param src_start_pos ut32, start position in source bitvector of copy
+ * \param dst RzBitVector, destination of copy (must be 64 or less bits)
+ * \param dst_start_pos ut32, start position in destination bitvector
+ * \param nbit ut32, control the size of copy (in bits)
+ * \return copied_size ut32, Actual copied size
+ */
+static ut32 rz_bv_copy_nbits_small(RZ_NONNULL const RzBitVector *src, ut32 src_start_pos, RZ_NONNULL RzBitVector *dst, ut32 dst_start_pos, ut32 nbit) {
+	// Sanity check performed by caller
+	if (nbit == 64) {
+		dst->bits.small_u = src->bits.small_u;
+	} else {
+		ut64 mask = ((1ull) << nbit) - 1;
+		dst->bits.small_u &= ~(mask << dst_start_pos);
+		dst->bits.small_u |= ((src->bits.small_u >> src_start_pos) & mask) << dst_start_pos;
```

Comment:
Please move this into a `static inline` in `rz_bits.h` and add 2-3 tests in `test_bits.c`. It is a common problem and a nice to have helper.